crates/client/rpc/src/errors.rs

```diff
@@ -208,3 +210,56 @@ impl From<StarknetApiError> for StarknetRpcApiError {
         StarknetRpcApiError::ErrUnexpectedError { data: err.to_string() }
     }
 }
+
+#[derive(Debug)]
+pub enum WsResult<'a> {
+    Ok,
+    Err(StarknetWsApiError<'a>),
+}
+
+#[cfg_attr(test, derive(PartialEq, Eq))]
+#[derive(Debug)]
+pub enum StarknetWsApiError<'a> {
+    TooManyBlocksBack,
+    Internal { context: Cow<'a, str> },
+}
+
+impl<'a> StarknetWsApiError<'a> {
+    fn code(&self) -> i32 {
+        match self {
+            Self::TooManyBlocksBack => 68,
+            Self::Internal { context: _ } => jsonrpsee::types::error::INTERNAL_ERROR_CODE,
+        }
+    }
+    fn message(&self) -> &str {
+        match self {
+            Self::TooManyBlocksBack => "Cannot go back more than 1024 blocks",
+            Self::Internal { context } => context,
```

Comment:
be careful, we do not want to give information to the client when there is an internal problem, only log the problem in madara

Reply:
Fixed.